lib/evaluate.js

```diff
@@ -279,6 +296,108 @@ const calculateInetGroupSuccessRates = (taskGroups) => {
   return result
 }
 
+/**
+ * @param {import('./typings').Measurement[]} measurements
+ * @param {import('./typings').Point} telemetryPoint
+ */
+const reportRetrievalStats = (measurements, telemetryPoint) => {
+  const totalCount = measurements.length
+  if (totalCount < 1) {
+    telemetryPoint.intField('measurements', totalCount)
+    telemetryPoint.intField('unique_tasks', 0)
+    return
+  }
+
+  const uniqueTasksCount = countUniqueTasks(measurements)
+
+  // Calculate aggregates per retrieval result
+  /** @type {Record<import('./typings').RetrievalResult, number> */
+  const resultBreakdown = {
+    OK: 0,
+    TIMEOUT: 0,
+    CAR_TOO_LARGE: 0,
+    BAD_GATEWAY: 0,
+    GATEWAY_TIMEOUT: 0
+  }
+
+  const participants = new Set()
+  const inetGroups = new Set()
+  let downloadBandwidth = 0
+
+  const ttfbHistogram = createHistogram()
+  const durationHistogram = createHistogram()
+  const sizeHistogram = createHistogram()
+
+  for (const m of measurements) {
+    const result = m.retrievalResult ?? 'UNDEFINED'
```

Comment:
is `UNDEFINED` missing in the `resultBreakdown` map?

Reply:
And when can this happen? Should it be an error we log?

Reply:
> is `UNDEFINED` missing in the `resultBreakdown` map?

I am intentionally not listing all possible keys in the initialiser of `resultBreakdown`, e.g. `ERROR_500` and `ERROR_404` are not listed either.

The idea is that if I don't explicitly include them with a `0` value in the initialiser, they won't show in our Grafana dashboard at all.

> And when can this happen? Should it be an error we log?

I don't have any specific case in mind; it's an _unkown unknown_. By mapping such measurements to `UNDEFINED ` bucket, we can notice them in our Grafana visualisations.

I can log this as an error, but I am concerned this could become very noisy - we are currently processing almost 400k measurements per round.

How about setting up a Grafana alert for the case when we find honest measurements with unknown retrieval results?

Reply:
Actually, `retrievalResult` should always be set by `lib/preprocess.js`, so this is pretty much about making the code more robust plus making it easier to debug failing unit tests that may forget to set `retrievalResult` property.